### ISSUE TITLE
Unable to load a previously saved snapshot

### ISSUE THREAD

**Issue by SteveFosdick** [stardot/b-em#94]
Having saved a snapshot, upon trying to load it, it is not shown in the file chooser.

**Comment by SteveFosdick** [comment]
The issue turned out to be that, when saving a snapshot the .snp file extension was not automatically added while the file chooser to load a snapshot had a filter to only show files with the .snp extension.

The solution was to add the .snp file extension on save when missing.

During testing I did come across a core dump loading a snapshot on the pam5000 branch but not on the issue94 branch (which is branched from master):

```
#0  0x00007ffff75ca79a in __strlen_sse2 () at /usr/lib/libc.so.6
#1  0x00007ffff7e8abf8 in al_ref_cstr () at /usr/lib/liballegro.so.5.2
#2  0x00007ffff7e1c225 in al_get_config_value () at /usr/lib/liballegro.so.5.2
#3  0x00005555555cf534 in get_config_string
    (sect=0x400000000 <error: Cannot access memory at address 0x400000000>, key=0x55555564cdc7 "os", sval=0x7fffe4368cf0 "mos352.rom") at config.c:70
#4  0x00005555555fd1e1 in mem_romsetup_master () at mem.c:226
#5  0x00005555555fe2d6 in model_init () at model.c:232
#6  0x00005555555fbb03 in main_restart () at main.c:314
#7  0x00005555555fea25 in model_loadstate (f=<optimized out>) at model.c:352
#8  0x0000555555603acc in load_state_two () at savestate.c:282
#9  0x0000555555603acc in savestate_doload () at savestate.c:352
#10 0x00005555555fc055 in main_timer (event=0x7fffffffe470) at main.c:439
#11 0x00005555555fc055 in main_run () at main.c:489
#12 0x000055555556d0ae in main (argc=<optimized out>, argv=<optimized out>)
    at main.c:583
```

**Comment by SteveFosdick** [comment]
Core dump resolved by commit 6fd707e8e3cec5b0f18635d9c33b12d08a4b7257
